File: Java/src/SearchInRotatedSortedArrayTest.java
import org.junit.Assert;
import org.junit.Test;

public class SearchInRotatedSortedArrayTest {

    @Test
    public void testSearch() {
        SearchInRotatedSortedArray s = new SearchInRotatedSortedArray();

        Assert.assertTrue(s.search(new int[]{}, 1) == -1);

        Assert.assertTrue(s.search(new int[]{0, 1, 2, 4, 5, 6, 7}, 7) == 6);

        Assert.assertTrue(s.search(new int[]{4, 5, 6, 7, 0, 1, 2}, 7) == 3);

        Assert.assertTrue(s.search(new int[]{5, 6, 7, 0, 1, 2, 4}, 7) == 2);

        Assert.assertTrue(s.search(new int[]{4, 5, 6, 7, 0, 1, 2}, 8) == -1);
    }

}

File: Kotlin/src/SearchInRotatedSortedArrayTest.kt
import org.junit.Test

class SearchInRotatedSortedArrayTest {

    @Test
    fun testSearch() {
        val s = SearchInRotatedSortedArray()

        assert(s.search(intArrayOf(), 1) == -1)

        assert(s.search(intArrayOf(0, 1, 2, 4, 5, 6, 7), 7) == 6)

        assert(s.search(intArrayOf(4, 5, 6, 7, 0, 1, 2), 7) == 3)

        assert(s.search(intArrayOf(5, 6, 7, 0, 1, 2, 4), 7) == 2)

        assert(s.search(intArrayOf(4, 5, 6, 7, 0, 1, 2), 8) == -1)
    }

}